includes/amp-helper-functions.php

```diff
@@ -139,7 +139,17 @@ function () {
 	 * version was new option added, and in that case default would never be used for a site
 	 * upgrading from a version prior to 1.0. So this is why get_option() is currently used.
 	 */
-	$options     = get_option( AMP_Options_Manager::OPTION_NAME, [] );
+	$options = get_option( AMP_Options_Manager::OPTION_NAME, [] );
+
+	// Initialize the plugin_configured option, setting it to true if options already exist.
+	if ( is_admin() && current_user_can( 'manage_options' )  ) {
+		if ( empty( $options ) ) {
+			AMP_Options_Manager::update_option( Option::PLUGIN_CONFIGURED, false );
+		} elseif ( ! isset( $options[ Option::PLUGIN_CONFIGURED ] ) ) {
+			AMP_Options_Manager::update_option( Option::PLUGIN_CONFIGURED, true );
+		}
+	}
+
```

Comment:
I think we can handle this with default values in `AMP_Options_Manager::get_options()`. I'll make the change.

Reply:
Go for it. I'll leave it alone.

Reply:
Does afaf9a6d6 do the trick?

Reply:
Not quite…

Reply:
That's what I tried at first but it didn't work, I think because of this: https://github.com/ampproject/amp-wp/blob/develop/includes/amp-helper-functions.php#L145-L160

Once `AMP_Options_Manager::update_option( Option::VERSION, AMP__VERSION );` is called, the options will no longer be empty, and that runs way before `get_options` is called again.

Reply:
That's why I was trying to squeeze in the initialization of `plugin_configured` before that point.

Reply:
It seems to be working for me.

If I reset the option to an old value:

```bash
echo '{"theme_support":"transitional","supported_post_types":["post"],"analytics":[],"all_templates_supported":false,"supported_templates":["is_singular"],"version":"1.5.5"}' | wp option set amp-options --format=json
```

Then:

```
$ wp eval 'var_dump(AMP_Options_Manager::get_option("plugin_configured"));'
bool(true)
```

Whereas if I delete the option entirely:

```
$ wp option delete amp-options
Success: Deleted 'amp-options' option.
```

Then I get:

```
$ wp eval 'var_dump(AMP_Options_Manager::get_option("plugin_configured"));'
bool(false)
```

Reply:
Hm, all right, I'll verify when I make the change for the settings page.